crates/chain/src/chain.rs

```
@@ -84,6 +86,33 @@ pub struct IrysNodeCtx {
     pub vdf_steps_guard: VdfStepsReadGuard,
     pub vdf_config: VDFStepsConfig,
     pub storage_config: StorageConfig,
+    // Shutdown channels
+    pub reth_shutdown_sender: tokio::sync::mpsc::Sender<()>,
+    // Thread handles spawned by the start function
+    pub reth_thread_handle: Option<CloneableJoinHandle<()>>,
+}
+
+impl IrysNodeCtx {
+    pub async fn stop(self) {
+        // Shutting down reth node will propagate to the main actor thread eventually
+        let chain_spec_arc = self.reth_handle.chain_spec();
+        let chain_spec = (*chain_spec_arc).clone();
+
+        self.reth_handle
+            .irys_ext
+            .as_ref()
+            .unwrap()
+            .reload
+            .write()
+            .unwrap()
+            .send(ReloadPayload::ReloadConfig(chain_spec))
+            .unwrap();
```

maybe early return / ignore the error? Because if a panic happens here then won't it stop the graceful shutdown logic in `chain.rs/fn start_reth_node`?

---

Hmm. I think that we actually should crash if we fail to send this message. All shutdown logic should be completed in a very sepcific order, and the only reason we can't send this message is because app's state got corrupted in some way

---

Removed that logic completely after removing reload channel